server/app/controllers/ti/TrustedIntermediaryController.java

```diff
@@ -100,6 +104,40 @@ public Result dashboard(
             messagesApi.preferred(request)));
   }
 
+  @Secure(authorizers = Authorizers.Labels.TI)
+  public Result updateDateOfBirth(Long accountId, Http.Request request) {
+    Optional<CiviFormProfile> civiformProfile = profileUtils.currentUserProfile(request);
+    if (civiformProfile.isEmpty()) {
+      return unauthorized();
+    }
+
+    Optional<TrustedIntermediaryGroup> trustedIntermediaryGroup =
+        userRepository.getTrustedIntermediaryGroup(civiformProfile.get());
+    if (trustedIntermediaryGroup.isEmpty()) {
+      return notFound();
+    }
+
+    Form<UpdateApplicantDob> form =
+        formFactory.form(UpdateApplicantDob.class).bindFromRequest(request);
+    Form<UpdateApplicantDob> returnedForm = null;
+
+    try {
+      returnedForm = tiService.updateApplicantDateOfBirth(accountId, form);
+    } catch (ApplicantNotFoundException e) {
+      throw new BadRequestException(e.getMessage());
```

Comment:
Actually this exception is used quite a bit in the Applicant Service, Applicant Repository and Applicant**Controllers... I am not very sure how the errors should be handled there if this exception is thrown. That would be a little beyond the scope of this PR. Other than modifying how this error is handled across the code base, i would throw 404 than 400.

Reply:
As I understand, all other callers would have to explicitly catch the exception since it's a checked exception. As such, adding it to ErrorHandler would mean that it would continue to work as intended here and we could slowly remove the explicit catches in other places that use it (similar to what we've been doing for ProgramNotFoundException).

I believe this should be safe to add to ErrorHandler without affecting the other callsites.

Reply:
Done! Thank you